src/app/(public)/repos/[language]/_components/sorter.tsx

```diff
@@ -129,7 +130,7 @@ export function Sorter() {
           <ul tabIndex={0} className="menu menu-vertical">
             {mainLanguages.sort(sortByName).map(language => (
               <li key={language} onClick={handleClick}>
-                <Link href={`/repos/${language.toLowerCase()}`}>
+                <Link href={`/repos/${language.toLowerCase()}?${urlSearchParam.toString()}`}>
```

Comment:
You can directly use the `searchParams.toString()` instead of creating another instance of it.

A new instance is created when the search params need to be modified.

Reply:
I see... Now I have fixed the issue. Please take a look.